pkg/resources/task.go

```diff
@@ -0,0 +1,478 @@
+		builder.WithSessionParameters(expandStringList(v.(*schema.Set).List()))
+	}
+
+	if v, ok := data.GetOk("user_task_timeout_ms"); ok {
+		builder.WithTimeout(v.(int))
+	}
+
+	if v, ok := data.GetOk("comment"); ok {
+		builder.WithComment(v.(string))
+	}
+
+	if v, ok := data.GetOk("after"); ok {
+		builder.WithDependency(v.(string))
+	}
+
+	if v, ok := data.GetOk("when"); ok {
+		builder.WithCondition(v.(string))
+	}
+
+	q := builder.Create()
+
+	err := snowflake.Exec(db, q)
+	if err != nil {
+		return errors.Wrapf(err, "error creating task %v", name)
+	}
+
+	taskID := &taskID{
+		DatabaseName: database,
+		SchemaName:   dbSchema,
+		TaskName:     name,
+	}
+	dataIDInput, err := taskID.String()
+	if err != nil {
+		return err
+	}
+	data.SetId(dataIDInput)
+
+	return ReadTask(data, meta)
+}
+
+// UpdateTask implements schema.UpdateFunc
+func UpdateTask(data *schema.ResourceData, meta interface{}) error {
+	// https://www.terraform.io/docs/extend/writing-custom-providers.html#error-handling-amp-partial-state
+	data.Partial(true)
+
+	db := meta.(*sql.DB)
+	taskID, err := taskIDFromString(data.Id())
+	if err != nil {
+		return err
+	}
+
+	database := taskID.DatabaseName
+	dbSchema := taskID.SchemaName
+	name := taskID.TaskName
+
+	builder := snowflake.Task(name, database, dbSchema)
+
+	if data.HasChange("warehouse") {
+		_, new := data.GetChange("warehouse")
+		q := builder.ChangeWarehouse(new.(string))
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating warehouse on task %v", data.Id())
+		}
+		data.SetPartial("warehouse")
+	}
+
+	if data.HasChange("schedule") {
+		var q string
+		old, new := data.GetChange("schedule")
+		if old != "" && new == "" {
+			q = builder.RemoveSchedule()
+		} else {
+			q = builder.ChangeSchedule(new.(string))
+		}
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating schedule on task %v", data.Id())
+		}
+		data.SetPartial("schedule")
+	}
+
+	if data.HasChange("user_task_timeout_ms") {
+		var q string
+		old, new := data.GetChange("user_task_timeout_ms")
+		if old.(int) > 0 && new.(int) == 0 {
+			q = builder.RemoveTimeout()
+		} else {
+			q = builder.ChangeTimeout(new.(int))
+		}
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating user task timeout on task %v", data.Id())
+		}
+		data.SetPartial("user_task_timeout_ms")
+	}
+
+	if data.HasChange("comment") {
+		var q string
+		old, new := data.GetChange("comment")
+		if old != "" && new == "" {
+			q = builder.RemoveComment()
+		} else {
+			q = builder.ChangeComment(new.(string))
+		}
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating comment on task %v", data.Id())
+		}
+		data.SetPartial("comment")
+	}
+
+	if data.HasChange("after") {
+		var (
+			q   string
+			err error
+		)
+		old, new := data.GetChange("after")
+
+		if old != "" {
+			q = builder.RemoveDependency(old.(string))
```

Comment:
It seems from the [docs](https://docs.snowflake.com/en/sql-reference/sql/alter-task.html) that removing will suspend the task and "must be restarted manually". Does adding a new `after` restart the task?

Reply:
Hmm, good catch! I'll take a look

Reply:
I'm resuming the task every time `after` changes now.